sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java

```diff
@@ -1072,6 +1071,37 @@ public void getStatesString() throws IOException {
     assertEquals("not found", result.stream().skip(1).findFirst().get().getError());
   }
 
+  @Test
+  public void getStateContext() throws IOException {
+    String etag = "ETag1";
+    String key = "key1";
+    String expectedValue = "hello world";
+    StateOptions options = buildStateOptions(StateOptions.Consistency.STRONG, StateOptions.Concurrency.FIRST_WRITE);
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("key_1", "val_1");
+    State<String> expectedState = buildStateKey(expectedValue, key, etag, new HashMap<>(), options);
+    DaprProtos.GetStateResponse responseEnvelope = DaprProtos.GetStateResponse.newBuilder()
+        .setData(serialize(expectedValue))
+        .setEtag(etag)
+        .build();
+    GetStateRequestBuilder builder = new GetStateRequestBuilder(STATE_STORE_NAME, key);
+    builder.withMetadata(metadata).withStateOptions(options);
+    GetStateRequest request = builder.build();
+    doAnswer((Answer<Void>) invocation -> {
+      StreamObserver<DaprProtos.GetStateResponse> observer = (StreamObserver<DaprProtos.GetStateResponse>) invocation.getArguments()[1];
+      observer.onNext(responseEnvelope);
+      observer.onCompleted();
+      return null;
+    }).when(daprStub).getState(any(DaprProtos.GetStateRequest.class), any());
+
+    Mono<State<String>> result = client.getState(request, TypeRef.STRING).subscriberContext(
+        it -> it.put("grpc-trace-bin", "foobar".getBytes(StandardCharsets.UTF_8))
```

Comment:
Where do you validate that "grpc-trace-bin" is being consumed?

Reply:
Removed this test case

Reply:
I actually liked it. Can this be validated?

Reply:
DaprClientGrpcTelemetryTest already validates subscriberContext. I don't think it's necessary (I wrote that test before finding DaprClientGrpcTelemetryTest)